pkg/cloud/services/network/subnets_test.go

```diff
@@ -1860,3 +2111,100 @@ func TestDeleteSubnets(t *testing.T) {
 		})
 	}
 }
+
+// Test helpers
```

Comment:
These test helpers using a builder pattern are excellent. Thanks for adding this.